### PR TITLE
feat: 'Typed Transaction Envelope' added to transaction page

### LINKED ISSUE #3807: Add new 'Typed Transaction Envelope' to transactions page
**Is your feature request related to a problem? Please describe.**
[EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) introduced Typed Transaction Envelope. 

**Describe the solution you'd like**
We should add Typed Transaction Envelope to [the transactions page](https://ethereum.org/en/developers/docs/transactions/)

**Additional context**
- https://eips.ethereum.org/EIPS/eip-2718
- https://blog.mycrypto.com/new-transaction-types-on-ethereum/

**Want to contribute?**

We love contributions from the Ethereum community! Please comment on an issue if you're interested in helping out with a PR.

### PR DESCRIPTION
**Pull request by MukulKolpe**
## Description

Added 'Typed Transaction Envelope' on the transaction page

References used:-
- https://eips.ethereum.org/EIPS/eip-2718
- https://blog.mycrypto.com/new-transaction-types-on-ethereum/

## Link to deploy preview

https://deploy-preview-3836--ethereumorg.netlify.app/en/developers/docs/transactions/

## Related Issue

Fixes #3807

### CODE REVIEW

**Comment by minimalsm** [comment]
Hey @MukulKolpe thanks for the PR on this. I know from experience it can be quite daunting taking up an issue then realising you know nothing about it and have to do some serious digging and learning for it to click 😅 Thanks for sticking with it and I hope you learned a lot from it.

**Comment by minimalsm** [comment]
@MukulKolpe just FYI I've pinged some experts to have a look before moving on this because I'm not familiar with typed transactions 🙂

File: src/content/developers/docs/transactions/index.md [hunk 1]
```diff
@@ -305,6 +305,22 @@ Watch Austin walk you through transactions, gas, and mining.
 
 <iframe width="100%" height="315" src="https://www.youtube.com/embed/er-0ihqFQB0" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
 
+## Typed Transaction Envelope
+
+Ethereum has different transaction types that specify what a transaction should do. For example, sending Ether to an address, deploying a contract, etc.  
+Ethereum now has a new transaction standard that was defined and created by Micah Zoltu in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718): Typed Transaction Envelope, which forms the basis for some new and other yet-to-be-developed features on Ethereum.
```
Comment (MicahZoltu) [inline src/content/developers/docs/transactions/index.md:311]: Recommend against using temporally sensitive terminology in documentation like "now has" and "that was".  The website should speak in terms of what is, largely ignoring what was unless explicitly discussing history.

Comment (MicahZoltu) [inline src/content/developers/docs/transactions/index.md:311]: Referring to the author of the EIP doesn't add value, so I recommend removing it here.  The EIP has an author line in the header if people are interested in that kind of history.

Comment (MicahZoltu) [reply src/content/developers/docs/transactions/index.md:311]: Recommend the following style, where the title is part of the link.
```
[EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718)
```

Comment (MicahZoltu) [inline src/content/developers/docs/transactions/index.md:311]: Recommend the following style, where the title is part of the link.
```
[EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718)
```

File: src/content/developers/docs/transactions/index.md [hunk 2]
```diff
@@ -305,6 +305,22 @@ Watch Austin walk you through transactions, gas, and mining.
...
+
+Previously, Ethereum had one format for transactions. Each transaction consists of a nonce, gas price, gas limit, to address, value, data, v, r, and s. These fields are RLP-encoded, to look something like this:
```
Comment (MicahZoltu) [inline src/content/developers/docs/transactions/index.md:313]: As above, recommend removing the temporally sensitive wording and focusing on the current state of things, which is that we have 4 transaction types now:
Cross Chain Legacy
Chain Locked Legacy [EIP-155](https://eips.ethereum.org/EIPS/eip-155)
Chain Locked with Access Lists [EIP-2930](https://eips.ethereum.org/EIPS/eip-2930)
Chain Locked with Split Fees [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)

File: src/content/developers/docs/transactions/index.md [hunk 3]
```diff
@@ -305,6 +305,22 @@ Watch Austin walk you through transactions, gas, and mining.
 
 <iframe width="100%" height="315" src="https://www.youtube.com/embed/er-0ihqFQB0" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
 
+## Typed Transaction Envelope
+
+Ethereum has different transaction types that specify what a transaction should do. For example, sending Ether to an address, deploying a contract, etc.
```
Comment (lightclient) [inline src/content/developers/docs/transactions/index.md:310]: This is a little inaccurate -- all tx types (so far) can send ether, deploy contracts, etc. These functions are considered different "types" of transactions. The different "types" have much more to do with allowing new transaction behavior that doesn't exist in the protocol already (e.g. access lists).

File: src/content/developers/docs/transactions/index.md (hunk 1 above)
Comment (lightclient) [reply src/content/developers/docs/transactions/index.md:311]: If *must* refer to the author of the EIP, I recommend using their full title.

```
Micah Zoltu, Overlord of Ethereum Governance
```

File: src/content/developers/docs/transactions/index.md [hunk 4]
```diff
@@ -305,6 +305,22 @@ Watch Austin walk you through transactions, gas, and mining.
...
 
+## Typed Transaction Envelope
+
+Ethereum has different transaction types that specify what a transaction should do. For example, sending Ether to an address, deploying a contract, etc.  
+Ethereum now has a new transaction standard that was defined and created by Micah Zoltu in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718): Typed Transaction Envelope, which forms the basis for some new and other yet-to-be-developed features on Ethereum.
+
+Previously, Ethereum had one format for transactions. Each transaction consists of a nonce, gas price, gas limit, to address, value, data, v, r, and s. These fields are RLP-encoded, to look something like this:  
+`RLP([nonce, gasPrice, gasLimit, to, value, data, v, r, s])`
+
+EIP-2718 defines a new generalised envelope for typed transactions. In the new standard, transactions look like this:
```
Comment (lightclient) [inline src/content/developers/docs/transactions/index.md:316]:
```suggestion
EIP-2718 defines a new generalised envelope for typed transactions. In the new standard, transactions are interpreted as:  
```

File: src/content/developers/docs/transactions/index.md (hunk 3 above)
Comment (MukulKolpe) [reply src/content/developers/docs/transactions/index.md:310]: Hey @lightclient, can you help me reframe or replace this statement?
Thanks!!!

Comment (lightclient) [reply src/content/developers/docs/transactions/index.md]: Maybe something like "Over time Ethereum has evolved to support multiple types of transactions. This allows new features like access lists and EIP-1559 to be implemented without affecting legacy transaction formats."?

Comment (MukulKolpe) [reply src/content/developers/docs/transactions/index.md]: yeah that looks great!!
Thanks !

File: src/content/developers/docs/transactions/index.md [hunk 5]
```diff
@@ -305,6 +305,22 @@ Watch Austin walk you through transactions, gas, and mining.
...
+`RLP([nonce, gasPrice, gasLimit, to, value, data, v, r, s])`
+
+EIP-2718 is a new generalised envelope for typed transactions. In the new standard, transactions are interpreted as:  
+`TransactionType || TransactionPayload`
+
+Where the fields are defined as:
+
+- TransactionType: a number between 0 and 0x7f, for a total of 128 possible transaction types.
+- TransactionPayload: an arbitrary byte array, defined by the transaction type.
+
```
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:323]:
```suggestion
## Typed Transaction Envelope {#typed-transaction-envelope}

Ethereum originally had one format for transactions. Each transaction consists of a nonce, gas price, gas limit, to address, value, data, v, r, and s. These fields are RLP-encoded, to look something like this:  

`RLP([nonce, gasPrice, gasLimit, to, value, data, v, r, s])`

Ethereum has evolved to support multiple types of transactions to allow for new features such as access lists and [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md) to be implemented without affecting legacy transaction formats.  

[EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718) defines a transaction type that is an envelope for future transaction types.

EIP-2718 is a new generalised envelope for typed transactions. In the new standard, transactions are interpreted as:  

`TransactionType || TransactionPayload`

Where the fields are defined as:

- `TransactionType` - a number between 0 and 0x7f, for a total of 128 possible transaction types.
- `TransactionPayload` an arbitrary byte array defined by the transaction type.

```

File: src/content/developers/docs/transactions/index.md [hunk 6]
```diff
@@ -305,6 +305,22 @@ Watch Austin walk you through transactions, gas, and mining.
...
 ## Further reading {#further-reading}
 
 _Know of a community resource that helped you? Edit this page and add it!_
```
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:326]:
```suggestion
## Further reading {#further-reading}

- [EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718) 

_Know of a community resource that helped you? Edit this page and add it!_
```

**Review by minimalsm — CHANGES_REQUESTED** [review]

File: src/content/developers/docs/transactions/index.md [hunk 7]
```diff
@@ -305,8 +305,29 @@ Watch Austin walk you through transactions, gas, and mining.
 
 <iframe width="100%" height="315" src="https://www.youtube.com/embed/er-0ihqFQB0" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
 
+## Typed Transaction Envelope {#typed-transaction-envelope}
+
+Ethereum originally had one format for transactions. Each transaction consists of a nonce, gas price, gas limit, to address, value, data, v, r, and s. These fields are RLP-encoded, to look something like this:
```
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:310]:
```suggestion
Ethereum originally had one format for transactions. Each transaction contained a nonce, gas price, gas limit, to address, value, data, v, r, and s. These fields are RLP-encoded, to look something like this:  
```

File: src/content/developers/docs/transactions/index.md [hunk 8]
```diff
@@ -305,8 +305,29 @@ Watch Austin walk you through transactions, gas, and mining.
...
+[EIP-2718: Typed Transaction Envelope](https://eips.ethereum.org/EIPS/eip-2718) defines a transaction type that is an envelope for future transaction types.
+
+EIP-2718 is a new generalised envelope for typed transactions. In the new standard, transactions are interpreted as:  
+
+`TransactionType || TransactionPayload`
+
+Where the fields are defined as:
+
+- `TransactionType` - a number between 0 and 0x7f, for a total of 128 possible transaction types.
+- `TransactionPayload` an arbitrary byte array defined by the transaction type.
```
Comment (minimalsm) [inline src/content/developers/docs/transactions/index.md:325]:
```suggestion
- `TransactionPayload` - an arbitrary byte array defined by the transaction type.
```

**Pull request merged**